test/core.jl

```diff
@@ -7295,6 +7295,24 @@ end
 struct T36104   # check that redefining it works, issue #21816
     v::Vector{T36104}
 end
+# with a gensymmed unionall
+struct Symmetric{T,S<:AbstractMatrix{<:T}} <: AbstractMatrix{T}
+    data::S
+    uplo::Char
+end
+struct Symmetric{T,S<:AbstractMatrix{<:T}} <: AbstractMatrix{T}
+    data::S
+    uplo::Char
+end
+@assert try
```

Comment:
Probably shouldn't use `@assert` for tests since `@assert` is documented to be allowed to be removed in some optimization levels and we probably want this to always check.